code/controllers/subsystem/ai_controllers.dm

```diff
@@ -24,10 +30,20 @@ SUBSYSTEM_DEF(ai_controllers)
 /datum/controller/subsystem/ai_controllers/stat_entry(msg)
 	var/list/active_list = ai_controllers_by_status[AI_STATUS_ON]
 	var/list/inactive_list = ai_controllers_by_status[AI_STATUS_OFF]
-	msg = "Active AIs:[length(active_list)]|Inactive:[length(inactive_list)]"
+	var/list/idle_list = ai_controllers_by_status[AI_STATUS_IDLE]
+	msg = "Active AIs:[length(active_list)]/[round(cost_on,1)]|Inactive:[length(inactive_list)]|Idle:[length(idle_list)]/[round(cost_idle,1)]"
 	return ..()
 
 /datum/controller/subsystem/ai_controllers/fire(resumed)
+	var/timer = TICK_USAGE_REAL
+	for(var/datum/ai_controller/ai_controller as anything in ai_controllers_by_status[AI_STATUS_IDLE])
+		for(var/client/client_found in GLOB.clients)
```

Comment:
We discussed how we should replace lavaland basic mob's oview with a proximity monitor, but I think oview is at least better than going through every single client man.

Reply:
this would be way more expensive, when lemon finishes his PR we can do that but we need a solution for the now and lemon's PR is a year old with change

Reply:
It's more expensive to check the turfs around each mob for something there than to check the distance between each mob and every individual client including ghosts?

Reply:
we could start mobs in idle and then de-idle them, perhaps

Reply:
mobs now start in idle instead of on

Reply:
> It's more expensive to check the turfs around each mob for something there than to check the distance between each mob and every individual client including ghosts?

yes. orange and oview process thru every atom in the range, even if done in the backend by byond the processing still happens.